packages/discord.js/src/structures/ThreadChannel.js

```diff
@@ -196,6 +208,16 @@ class ThreadChannel extends Channel {
     return this.archiveTimestamp && new Date(this.archiveTimestamp);
   }
 
+  /**
+   * The time the thread was created at
+   * @type {?Date}
+   * @readonly
+   */
+  get createdAt() {
+    // Private threads can rely on their ID for creation date
+    return (this.createdTimestamp ??= this.type === ChannelType.PrivateThread ? super.createdTimestamp : null);
```

Comment:
This is returning a number, not a date, also it's... assigning? I don't think we have a getter that does side-effects, and it feels wrong to be honest.